cmd/aks-periscope/aks-periscope.go

```diff
@@ -55,35 +105,104 @@ func main() {
 	osmCollector := collector.NewOsmCollector()
 	smiCollector := collector.NewSmiCollector()
 
-	collectors := []interfaces.Collector{
-		containerLogsCollector,
-		dnsCollector,
-		kubeObjectsCollector,
-		networkOutboundCollector,
+	selectedCollectors := selectCollectors(
+		map[string]interfaces.Collector{
+			containerLogsCollector.GetName():           containerLogsCollector,
+			systemLogsCollector.GetName():              systemLogsCollector,
+			networkOutboundCollector.GetName():         networkOutboundCollector,
+			ipTablesCollector.GetName():                ipTablesCollector,
+			nodeLogsCollector.GetName():                nodeLogsCollector,
+			dnsCollector.GetName():                     dnsCollector,
+			kubeObjectsCollector.GetName():             kubeObjectsCollector,
+			kubeletCmdCollector.GetName():              kubeletCmdCollector,
+			systemPerfCollector.GetName():              systemPerfCollector,
+			helmCollector.GetName():                    helmCollector,
+			osmCollector.GetName():                     osmCollector,
+      			smiCollector.GetName():                     smiCollector,
+		})
+
+	//diagnosers
+	//NOTE currently the collector instance objects are shared between the collector itself and things which use it as a dependency
+	networkConfigDiagnoser := diagnoser.NewNetworkConfigDiagnoser(dnsCollector, kubeletCmdCollector)
+	networkOutboundDiagnoser := diagnoser.NewNetworkOutboundDiagnoser(networkOutboundCollector)
+	selectedDiagnosers := selectDiagnosers(
+		map[string]interfaces.Diagnoser{
+			networkConfigDiagnoser.GetName():   networkConfigDiagnoser,
+			networkOutboundDiagnoser.GetName(): networkOutboundDiagnoser,
+		})
+
+	//exporters
+	azureBlobExporter := exporter.NewAzureBlobExporter(creationTimeStamp, hostname)
+	selectedExporters := selectExporters(
+		map[string]interfaces.Exporter{
+			azureBlobExporter.GetName():    azureBlobExporter,
+		})
+
+	return selectedCollectors, selectedDiagnosers, selectedExporters
+}
+
+// selectCollectors select the collectors to run by looking up the value from the config
+func selectCollectors(allCollectorsByName map[string]interfaces.Collector) []interfaces.Collector {
+	collectors := []interfaces.Collector{}
+
+	enabledCollectorString, found := os.LookupEnv("ENABLED_COLLECTORS")
+	if !found {
+		//if not defined, default to all collectors enabled
+		enabledCollectorString = "containerlogs dns helm iptables kubeletcmd kubeobjects networkoutbound nodelogs osm smi systemlogs systemperf"
```

Comment:
💡 If this is `default list` of collectors, then this is incorrect because `osm, smi` both are by conditional check checks, which are based one `OSM` as value in collectorList

Reply:
CollectorList value of OSM currently selects both OSM and SMI collectors to run. Value of SMI selects just SMI collector. Intention is for default behaviour being to run ALL collectors

Reply:
for current value of collectorlist = OSM see https://github.com/Azure/aks-periscope/blob/a1bdc52496b5c05547236e2cc7c4d3b15cd1359f/cmd/aks-periscope/aks-periscope.go#L76

Reply:
Well, see the two lines under that one 🙂

Reply:
We should try remove the need to update this string as a requirement of adding a new collector. Should dovetail well with the change to use a proper type registry mentioned in initial PR description.